### PR TITLE
[WIP]: Batch relayer support for stETH

### PR DESCRIPTION
**Pull request by evgenyboxer**
# Description

Adds batch relayer support for trading stETH.

Addresses for Mainnet will need to be provided before merging. (and SOR support for getting rates for stETH)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Wrap/unwrap stETH
- [ ] Trade ETH <> stETH
- [ ] Trade ERC20 <> stETH
- [ ] There should only be one approval for the batch relayer.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] My changes generate no new console warnings
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

File: src/composables/queries/useAllowancesQuery.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ type QueryResponse = ContractAllowancesMap;
  */
 export default function useAllowancesQuery(
   tokens: Ref<TokenInfoMap> = ref({}),
-  contractAddesses: Ref<string[]> = ref([]),
+  contractAddresses: Ref<string[]> = ref([]),
```
Comment (evgenyboxer) [inline src/composables/queries/useAllowancesQuery.ts:21]: This was just a typo

File: src/composables/queries/useBatchRelayerApprovalQuery.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      account.value,
+      configService.network.addresses.batchRelayer
+    );
+
+    return approved;
+  };
+
+  const queryOptions = reactive({
+    enabled,
+    ...FETCH_ONCE_OPTIONS,
```
Comment (evgenyboxer) [inline src/composables/queries/useBatchRelayerApprovalQuery.ts:58]: No need to refetch it more than once

File: src/composables/useTransactions.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ import { GnosisTransactionDetails } from './trade/useGnosis';
 
 const WEEK_MS = 86_400_000 * 7;
 // Please update the schema version when making changes to the transaction structure.
-const TRANSACTIONS_SCHEMA_VERSION = '1.1.1';
+const TRANSACTIONS_SCHEMA_VERSION = '1.1.2';
```
Comment (evgenyboxer) [inline src/composables/useTransactions.ts:25]: I have changed the schema for approval transaction - so I'm bumping it. (`tokenAddress` -> `contractAddress`)

File: src/locales/default.json [hunk 4]
```diff
@@ -17,6 +17,9 @@
     "amountBeforeFees": "Amount before fees",
     "amplification": "Amplification",
     "approve": "Approve",
+    "approveBatchRelayer": "Approve batch relayer for stETH",
+    "approvedBatchRelayer": "Approved batch relayer for stETH",
```
Comment (evgenyboxer) [inline src/locales/default.json:21]: the "for stETH" part just makes it clearer why is this extra step needed.

File: src/providers/tokens.provider.ts [hunk 5]
```diff
@@ -95,15 +105,12 @@ export default {
...
-        ...(networkConfig.addresses.exchangeProxy
-          ? [networkConfig.addresses.exchangeProxy]
-          : []),
-        ...(GP_ALLOWANCE_MANAGER_CONTRACT_ADDRESS
-          ? [GP_ALLOWANCE_MANAGER_CONTRACT_ADDRESS]
-          : [])
-      ]
+        networkConfig.addresses.wstETH,
+        networkConfig.addresses.exchangeProxy,
+        GP_ALLOWANCE_MANAGER_CONTRACT_ADDRESS
```
Comment (evgenyboxer) [inline src/providers/tokens.provider.ts:112]: @garethfuller I assume you wanted to remove these addresses if they are "empty" right?
If so, lodash `compact` comes in handy for these types of situations.

File: src/components/forms/pool_actions/InvestForm.vue [hunk 6]
```diff
@@ -506,8 +534,8 @@ export default defineComponent({
         : [isPositive()];
     }
 
-    function symbolFor(token) {
-      return tokens.value[token]?.symbol || '';
+    function symbolFor(token: string) {
+      return tokens.value[getAddress(token)]?.symbol || '';
```
[the remark this answers is not in the document]
Comment (garethfuller) [reply src/components/forms/pool_actions/InvestForm.vue]: Checked and it wasn't needed, so I've reverted here: https://github.com/balancer-labs/frontend-v2/pull/647/commits/3a764d533921937323bd81b0c37badda0ae11eaf let me know if I've missed something.

**Review by garethfuller — APPROVED** [review]

**Pull request merged**